plugins/database/hana/hana_test.go

```diff
@@ -22,122 +23,240 @@ func TestHANA_Initialize(t *testing.T) {
+				},
+				Password: test.startingPassword,
+				Statements: dbplugin.Statements{
+					Commands: []string{testHANARole},
+				},
+				Expiration: time.Now().Add(time.Hour),
+			}
+
+			userResp := dbtesting.AssertNewUser(t, db, newReq)
+			assertCredsExist(t, connURL, userResp.Username, test.startingPassword)
+
+			test.req.Username = userResp.Username
+
+			dbtesting.AssertUpdateUser(t, db, test.req)
+			err := testCredsExist(t, connURL, userResp.Username, test.req.Password.NewPassword)
+			if test.expectErrOnLogin {
+				if err == nil {
+					t.Fatalf("Able to login with new creds when expecting an issue")
+				} else if test.expectedErrMsg != "" && !strings.Contains(err.Error(), test.expectedErrMsg) {
+					t.Fatalf("Expected error message to contain \"%s\", received: %s", test.expectedErrMsg, err)
+				}
+			}
+			if !test.expectErrOnLogin && err != nil {
+				t.Fatalf("Unable to login: %s", err)
+			}
+		})
 	}
+}
 
-	// Test custom revoke statement
-	username, password, err = db.CreateUser(context.Background(), statements, usernameConfig, time.Now().Add(time.Hour))
-	if err != nil {
-		t.Fatalf("err: %s", err)
+func TestHANA_DeleteUser(t *testing.T) {
+	if os.Getenv("HANA_URL") == "" || os.Getenv("VAULT_ACC") != "1" {
+		t.SkipNow()
 	}
-	if err = testCredsExist(t, connURL, username, password); err != nil {
-		t.Fatalf("Could not connect with new credentials: %s", err)
+	connURL := os.Getenv("HANA_URL")
+
+	type testCase struct {
+		req              dbplugin.DeleteUserRequest
 	}
 
-	statements.Revocation = []string{testHANADrop}
-	err = db.RevokeUser(context.Background(), statements, username)
-	if err != nil {
-		t.Fatalf("err: %s", err)
+	tests := map[string]testCase{
+		"no update statements": {
+			req: dbplugin.DeleteUserRequest{},
+		},
+		"with custom update statements": {
+			req: dbplugin.DeleteUserRequest{
+				Statements: dbplugin.Statements{
+					Commands: []string{testHANADrop},
+				},
+			},
+		},
 	}
-	if err := testCredsExist(t, connURL, username, password); err == nil {
-		t.Fatal("Credentials were not revoked")
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			connectionDetails := map[string]interface{}{
+				"connection_url": connURL,
+			}
+
+			initReq := dbplugin.InitializeRequest{
+				Config:           connectionDetails,
+				VerifyConnection: true,
+			}
+
+			db := new()
+			dbtesting.AssertInitialize(t, db, initReq)
+			defer dbtesting.AssertClose(t, db)
+
+			password := "this_is_Thirty_2_characters_wow_"
+
+			newReq := dbplugin.NewUserRequest{
+				UsernameConfig: dbplugin.UsernameMetadata{
+					DisplayName: "test-test",
+					RoleName:    "test-test",
+				},
+				Password: password,
+				Statements: dbplugin.Statements{
+					Commands: []string{testHANARole},
+				},
+				Expiration: time.Now().Add(time.Hour),
+			}
+
+			userResp := dbtesting.AssertNewUser(t, db, newReq)
+			assertCredsExist(t, connURL, userResp.Username, password)
+
+			test.req.Username = userResp.Username
```

Comment:
Instead of manipulating the test case, what do you think about making the test case have a list of deletion statements rather than the full request object? That way you can construct the request here without having it be a part of the test case.